test/query/query.test.ts

```diff
@@ -97,3 +99,77 @@ describe('query/query', () => {
     });
   });
 });
+
+describe('2D aggregate', () => {
+  describe('summary with mean', () => {
+    it('should not generate 2D summary plot with MEAN(Q) x MEAN(Q1)', () => {
+      const q = {
+        spec: {
+          mark: Mark.POINT,
+          encodings:[
+            {
+              channel: Channel.X,
+              bin: SHORT_ENUM_SPEC,
+              aggregate: SHORT_ENUM_SPEC,
+              field: 'Q',
+              type: Type.QUANTITATIVE
+            },
+            {
+              channel: Channel.Y,
+              bin: SHORT_ENUM_SPEC,
+              aggregate: SHORT_ENUM_SPEC,
+              field: 'Q1',
+              type: Type.QUANTITATIVE
+            }
+          ],
+        },
+        nest: [{
+          groupBy: [Property.FIELD, Property.AGGREGATE, Property.BIN, Property.TIMEUNIT]
+        }],
+        config: {
+          autoAddCount: true,
+          omitAggregatePlotWithoutDimension: true
+        }
+      };
+      const CONFIG_WITH_OMIT_AGGREGATE_PLOT_WITHOUT_DIMENSION = extend({}, DEFAULT_QUERY_CONFIG, {omitAggregatePlotWithoutDimension: true});
+      const result = query(q, schema, CONFIG_WITH_OMIT_AGGREGATE_PLOT_WITHOUT_DIMENSION).result;
+      assert.equal(result.items.length, 6);
+    });
+  });
+
+    it('should generate 2D summary plot with MEAN(Q) x MEAN(Q1)', () => {
+      const q = {
+        spec: {
+          mark: Mark.POINT,
+          encodings:[
+            {
+            channel: Channel.X,
+            bin: SHORT_ENUM_SPEC,
+            aggregate: SHORT_ENUM_SPEC,
+            field: 'Q',
+            type: Type.QUANTITATIVE
+            },
+            {
+            channel: Channel.Y,
+            bin: SHORT_ENUM_SPEC,
+            aggregate: SHORT_ENUM_SPEC,
+            field: 'Q1',
+            type: Type.QUANTITATIVE
+            }
+          ],
+        },
+        nest: [{
+          groupBy: [Property.FIELD, Property.AGGREGATE, Property.BIN, Property.TIMEUNIT]
+        }],
+        config: {
+          autoAddCount: true,
+          omitAggregatePlotWithoutDimension: false
+        }
+      };
+      const CONFIG_WITH_OMIT_AGGREGATE_PLOT_WITHOUT_DIMENSION = extend({}, DEFAULT_QUERY_CONFIG, {omitAggregatePlotWithoutDimension: false});
```

Comment:
I don't understand why you need to add this, which is not only unnecessary but also increase the chance that you'll forget things.